File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

int solution(vector<vector<int>> data, int col, int row_begin, int row_end) {
    int answer = 0;
    
    sort(data.begin(), data.end(), [col](vector<int> a, vector<int> b){
        if (a[col - 1] == b[col - 1])
            return a[0] > b[0];
        
        return a[col - 1] < b[col - 1];
         });
    
    for(int i = row_begin-1; i <= row_end-1; ++i){
        int sum = 0;
        
        for(int j = 0; j < data[0].size(); ++j)
        {
            sum += data[i][j] % (i + 1);
        }
        
        
        answer = answer^sum;
    }
    
    return answer;
}

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.java

import java.util.Arrays;
import java.util.Comparator;

public class Solution {

    public static int solution(int[][] data, int col, int row_begin, int row_end) {
        int answer = 0;

        Arrays.sort(data, new Comparator<int[]>() {
            @Override
            public int compare(int[] o1, int[] o2) {
                
                if (o1[col - 1] == o2[col - 1])
                    return o2[0] - o1[0];

                return o1[col - 1] - o2[col - 1];
            }
        });

        int[] nums = new int[row_end - row_begin + 1];
        int idx = 0;
        for (int i = row_begin - 1; i <= row_end - 1; i++) {
            for (int j = 0; j < data[i].length; j++) {
                nums[idx] += data[i][j] % (i + 1);
            }
            idx++;
        }


        answer = nums[0];
        for (int i = 1; i < nums.length; i++) {
            answer ^= nums[i];
        }
        return answer;
    }
}